extensions/amp-next-page/0.2/page.js

```diff
@@ -0,0 +1,249 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {PositionInViewportEntryDef} from '../../../src/service/position-observer/position-observer-worker';
+import {RelativePositions} from '../../../src/layout-rect';
+import {VisibilityState} from '../../../src/visibility-state';
+import {dev} from '../../../src/log';
+
+/** @enum {string} */
+export const PageState = {
+  QUEUED: 'queued',
+  FETCHING: 'fetching',
+  LOADED: 'loaded',
+  FAILED: 'failed',
+  INSERTED: 'inserted',
+};
+
+/** @enum {string} */
+export const PageRelativePos = {
```

Comment:
Do these need to be string enums (i.e. you need to get a string from an external source and compare it to an enum entry)? I'm not sure I see a case where they can't just be numbers.